### PR TITLE
feat(ui5-badge): update design

### PR DESCRIPTION
**Pull request by gmkv**
The Fiori 3 specification updated the design of the Badge component.
Fixed a bug where the icon can get very small if the text is short.

Fixes: #3389
Fixes: #3518

### CODE REVIEW

File: packages/main/src/Badge.js [hunk 1]
```diff
@@ -114,11 +116,8 @@ class Badge extends UI5Element {
 	}
 
 	onBeforeRendering() {
-		if (this.hasIcon) {
-			this.setAttribute("__has-icon", "");
-		} else {
-			this.removeAttribute("__has-icon");
-		}
+		this.toggleAttribute("__has-icon", this.hasIcon);
```
Comment (ilhan007) [inline packages/main/src/Badge.js:119]: I think toggleAttribute is not supported in IE. And we should keep IE working until the end of the year.

Comment (gmkv) [reply packages/main/src/Badge.js]: Removed it.

File: packages/main/src/themes/Badge.css [hunk 2]
```diff
@@ -2,28 +2,49 @@
 
 :host(:not([hidden])) {
 	display: inline-block;
-	height: 1.125em;
+	height: 1em;
```
Comment (dimovpetar) [inline packages/main/src/themes/Badge.css:5]: better use rem to be consistent with other components

**Review by dimovpetar — CHANGES_REQUESTED** [review]

File: packages/main/src/themes/Badge.css (hunk 2 above)
Comment (gmkv) [reply packages/main/src/themes/Badge.css:5]: Ems are used here to enable scaling of the icon based on the font size #729

File: packages/main/src/Badge.js [hunk 3]
```diff
@@ -119,6 +121,12 @@ class Badge extends UI5Element {
 		} else {
 			this.removeAttribute("__has-icon");
 		}
+
+		if (this.hasIconOnly) {
+			this.setAttribute("__has-icon-only", "");
+		} else {
+			this.removeAttribute("__has-icon-only");
+		}
```
[the remark this answers is not in the document]
Comment (gmkv) [reply packages/main/src/Badge.js]: Done

File: packages/main/src/Badge.js [hunk 4]
```diff
@@ -32,6 +32,22 @@ const metadata = {
 			type: String,
 			defaultValue: "1",
 		},
+
+		/**
+		 * Defines if the badge has an icon.
+		 * @private
+		 */
+		__hasIcon: {
```
Comment (fifoosid) [inline packages/main/src/Badge.js:40]: Do we really need two underlines in the beginning. We usually use only 1

File: packages/main/src/Badge.js [hunk 5]
```diff
@@ -32,6 +32,22 @@ const metadata = {
...
+			type: Boolean,
+		},
+
+		/**
+		 * Defines if the badge has only an icon (and no text).
+		 * @private
+		 */
+		__hasIconOnly: {
```
Comment (fifoosid) [inline packages/main/src/Badge.js:48]: same here 
Moreover isIconOnly or just iconOnly is more exact

File: packages/main/src/Badge.js [hunk 6]
```diff
@@ -129,6 +142,10 @@ class Badge extends UI5Element {
 		return !!this.icon.length;
 	}
 
+	get hasIconOnly() {
```
Comment (fifoosid) [inline packages/main/src/Badge.js:145]: isIconOnly or iconOnly as well

**Review by fifoosid — CHANGES_REQUESTED** [review]

File: packages/main/src/Badge.js (hunk 6 above)
Comment (gmkv) [reply packages/main/src/Badge.js]: Done

File: packages/main/src/Badge.js (hunk 5 above)
Comment (gmkv) [reply packages/main/src/Badge.js]: Done

File: packages/main/src/Badge.js (hunk 4 above)
Comment (gmkv) [reply packages/main/src/Badge.js]: Done

**Review by dimovpetar — APPROVED** [review]

**Review by fifoosid — APPROVED** [review]

**Pull request merged**